dubbo-config/dubbo-config-spring/src/main/java/org/apache/dubbo/config/spring/ReferenceBean.java

```diff
@@ -443,6 +445,11 @@ private Object getCallProxy() throws Exception {
             return referenceConfig.get();
         }
         synchronized (LockUtils.getSingletonMutex(applicationContext)) {
+            if (SCOPE_LOCAL.equals(referenceConfig.getScope())) {
+                // In some case, local(InJVM) providers are not exported when consumer calls
+                // like user call a local service when bean initializing
+                tryExportLocalProvider();
```

Comment:
Export the service here is not a good idea. 
I think there might not exist an elegant resolution for such scenario.